### PR TITLE
Move CI runtime logic from pipeline to ci_integration.sh

### PR DESCRIPTION
**Pull request by adrienthebo**

### CODE REVIEW

File: test/ci_integration.sh [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+# Map the input parameters provided by Concourse CI, or whatever mechanism is
+# running the tests to Terraform input variables.  Also setup credentials for
+# use with kitchen-terraform, inspec, and gcloud.
+setup_environment() {
+  local tmpfile
+  tmpfile="$(mktemp)"
+  echo "${SERVICE_ACCOUNT_JSON}" > "${tmpfile}"
+
+  # TODO: remove `credentials_path_relative` from test modules in favor of
```
Comment (aaron-lane) [inline test/ci_integration.sh]: TODOs should be filed as issues.

File: test/ci_integration.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  export TMPDIR="${DELETE_AT_EXIT}"
+  trap finish EXIT
+
+  # Setup environment variables
+  setup_environment
+  set -x
+
+  # Execute the test lifecycle
+  kitchen create
+  kitchen converge
```
Comment (aaron-lane) [inline test/ci_integration.sh]: The second converge present in the Makefile was dropped. Was it not necessary?

File: test/ci_integration.sh [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  export CLOUDSDK_AUTH_CREDENTIAL_FILE_OVERRIDE="${tmpfile}"
+  # Application default credentials (Terraform google provider and inspec-gcp)
+  export GOOGLE_APPLICATION_CREDENTIALS="${tmpfile}"
+
+  # Terraform variables
+  export TF_VAR_project_id="$PROJECT_ID"
+  export TF_VAR_credentials_path_relative="../shared/credentials.json"
+  export TF_VAR_region="$REGION"
+  export TF_VAR_zones="$ZONES"
+  export TF_VAR_compute_engine_service_account="$COMPUTE_ENGINE_SERVICE_ACCOUNT"
```
Comment (aaron-lane) [inline test/ci_integration.sh]: Does `compute_engine_service_account` need to be provided to the Docker containers as well?

Comment (adrienthebo) [reply test/ci_integration.sh]: Good catch, amended in d2f60e9

File: test/ci_integration.sh (hunk 2 above)
Comment (adrienthebo) [reply test/ci_integration.sh]: I'm not certain; for the sake of consistency with the old pipeline script let's just do the double converge. Amended in d2f60e9.

File: examples/deploy_service/main.tf [hunk 4]
```diff
@@ -21,6 +21,7 @@ locals {
 provider "google" {
   credentials = "${file(var.credentials_path)}"
   region      = "${var.region}"
+  version     = "~> 1.20"
```
Comment (Jberlinsky) [inline examples/deploy_service/main.tf]: These version-pins seem like a separate issue to me. Thoughts on extracting to a separate PR?

Comment (adrienthebo) [reply examples/deploy_service/main.tf]: Agreed, extracted to https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/90

**Comment by Jberlinsky** [comment]
Although CI is red, this PR is necessary to move forward, and the CI failures are not exclusively due to this PR. I'm signing off on merging this over the CI failure, unless @morgante and/or @aaron-lane strongly object.

**Pull request merged**